compiler/rustc_trait_selection/src/traits/object_safety.rs

```diff
@@ -941,5 +941,10 @@ pub fn contains_illegal_impl_trait_in_trait<'tcx>(
 }
 
 pub fn provide(providers: &mut Providers) {
-    *providers = Providers { object_safety_violations, check_is_object_safe, ..*providers };
+    *providers = Providers {
+        object_safety_violations,
+        check_is_object_safe,
+        generics_require_sized_self,
```

Comment:
Perhaps should we convert the existing usages of `generics_require_sized_self` to go thru tcx for caching purposes?